panel_segmentation/utils.py

```diff
@@ -0,0 +1,709 @@
+        for top in range(0, img_height, pixel_meter_conversion):
+            for left in range(0, img_width, pixel_meter_conversion):
+
+                # Create a Window and calculate the transform
+                window = rasterio.windows.Window(left, top,
+                                                 pixel_meter_conversion,
+                                                 pixel_meter_conversion)
+                transform = img.window_transform(window)
+
+                # Skip images where all pixels are black
+                if np.all(img.read(window=window) == 0):
+                    continue
+
+                # Create a new cropped raster to write to
+                profile = img.profile
+                profile.update({'height': pixel_meter_conversion,
+                                'width': pixel_meter_conversion,
+                                'transform': transform,
+                                'driver': "PNG"})
+                # Get center of cropped image
+                center = pixel_meter_conversion/2
+                # Check if crs is in  "EPSG:4326" lat-lon coordinates.
+                # If it's not, transform it to "EPSG:4326" coordinates.
+                # Otherwise, get coordinates directly
+                lon, lat = rasterio.transform.xy(transform, center, center)
+                if img.crs and not img.crs.is_geographic:
+                    transformer = Transformer.from_crs(img.crs, "EPSG:4326",
+                                                       always_xy=True)
+                    lon, lat = transformer.transform(lon, lat)
+
+                # Put coordinats in lat_lon format for file name
+                lat_lon = f"{lat:.7f}_{lon:.7f}"
+                file_path = os.path.join(file_save_folder, f"{lat_lon}.png")
+                with rasterio.open(file_path, 'w',
+                                   **profile) as png:
+                    # Read the data from the window and write it as a
+                    # png output
+                    png.write(img.read(window=window))
+    return
+
+
+def locateLatLonGeotiff(geotiff_file, latitude, longitude,
+                        file_name_save, pixel_resolution=300):
+    """
+    Locate a lat-lon coordinate in a GEOTIFF image, and then box that area
+    and capture a PNG image of it.
+
+    Parameters
+    -----------
+    geotiff_file: str
+        File name of the TIF file we want to scan for a particular latitude-
+        longitude coordinate.
+    latitude: float
+        Target latitude coordinate we want to find in the TIFF file.
+    longitude: float
+        Target longitude coordinate we want to find in the TIFF file.
+    file_name_save: str
+        Name of the file of the image taken of the target lat-lon
+        coordinate and its surrounding area.
+    pixel_resolution : int, default 300
+        Number of pixels in the x- and y-direction of the resulting image.
+
+    Returns
+    -------
+    image or None
```

Comment:
Add  description of image: returned image with image center set as designated lat-lon, or something like that